inference-engine/src/mkldnn_plugin/nodes/mkldnn_strided_slice_node.cpp

```diff
@@ -0,0 +1,516 @@
+
+    SizeVector beginDims = stridedSliceLayer->insData[BEGIN_ID].lock()->getTensorDesc().getDims();
+    if (beginDims.size() != 1)
+        THROW_ERROR << " should have begin vector with 1 dimension";
+
+    SizeVector endDims = stridedSliceLayer->insData[END_ID].lock()->getTensorDesc().getDims();
+    if (endDims.size() > 1)
+        THROW_ERROR << "should have end vector with 1 dimension";
+    if (beginDims[0] != endDims[0])
+        THROW_ERROR << "should have begin vector with size equal to end vector size";
+
+    SizeVector strideDims;
+    if (stridedSliceLayer->insData.size() > 3) {
+        strideDims = stridedSliceLayer->insData[STRIDE_ID].lock()->getTensorDesc().getDims();
+        if (strideDims.size() > 1)
+            THROW_ERROR << "should have stride vector with 1 dimension";
+        if (beginDims[0] != strideDims[0])
+            THROW_ERROR << "should have stride vector with size equal to begin vector size";
+    }
+
+    auto createMask = [&](const char* maskName, std::vector<int>& mask, const int bit) {
+        std::string::size_type i;
+        std::string maskStr = stridedSliceLayer->GetParamAsString(maskName, "");
+        for (i = 0; i < maskStr.size(); ++i) {
+            if (maskStr[i] == '1')
+                mask.push_back(1);
+            else if (maskStr[i] == '0')
+                mask.push_back(0);
+            else if (maskStr[i] != ',')
+                THROW_ERROR << "has incorrect '" << maskName << "'. Mask values should be 0 or 1";
+        }
+        if (strcmp(maskName, "ellipsis_mask") != 0 || i == 0) {
+            for (; i < srcDims.size(); ++i) mask.push_back(bit);
+        }
+    };
+
+    createMask("begin_mask", beginMask, 1);
+    createMask("end_mask", endMask, 1);
+    createMask("new_axis_mask", newAxisMask, 0);
+    createMask("shrink_axis_mask", shrinkAxisMask, 0);
+    createMask("ellipsis_mask", ellipsisMask, 0);
+
+    int ellipsisMaskCounter = 0;
+    int ellipsisPos1 = -1;
+    for (size_t i = 0; i < ellipsisMask.size(); i++) {
+        ellipsisMaskCounter += ellipsisMask[i];
+        ellipsisPos1 = ellipsisMask[i] == 1 && ellipsisPos1 == -1 ? i : ellipsisPos1;
+    }
+    if (ellipsisMaskCounter > 1)
+        THROW_ERROR << "has incorrect 'Ellipsis_mask'. Only one non-zero bit is allowed";
+
+    int newAxis = 0;
+    for (auto na : newAxisMask)
+        newAxis += na;
+    size_t maxDims = srcDims.size() + newAxis;
+
+    int shrinkAxis = 0;
+    for (auto sa : shrinkAxisMask)
+        shrinkAxis += sa;
+    params.equalDims = srcDims.size() == maxDims && shrinkAxis == 0;
+
+    auto fillingInParameters = [&](std::vector<int>& parameter, const size_t type, const size_t size, const int bit) {
+        auto blob = dynamic_cast<TBlob<int>*>(getParentEdgesAtPort(type)[0]->getParent()->getCnnLayer()->blobs["custom"].get());
+        int* ptr = blob->buffer().as<int*>() + blob->getTensorDesc().getBlockingDesc().getOffsetPadding();
```

Comment:
const int* ptr = blob->cbuffer().as<const int*>()...

Reply:
Fixed, thank you